concepts/type-theory-glossary.md

```diff
@@ -0,0 +1,396 @@
+if it has the relevant methods, then you can start using values of that type wherever the interface is required.
+
+In Rust, the `trait` feature combined with the `impl`+`for` keywords is most comparable to the concept of interface in other languages.
+Implementing a trait is [*explicit*](#explicit-versus-structural-interfaces) -- a type must use the `impl`+`for` keywords to explicitly state the methods that satisfy the trait.
+
+### what about numbers?
+
+Numbers are... interesting.
+
+For the most part in computing, we tend to pretend and build abstractions as if
+numbers are [infinite scalars](#infinite-scalars).
+
+Of course, this is rarely true in practice.
+Integers are often implemented in such a way that values that would require beyond 32 (or 64, etc) bits are unrepresentable;
+arguably, this makes them more like a (very (*very*) large) enum than anything else.
+Floating point numbers implemented in IEE764 style also have their own entire host of foibles.
+
+The scope of this document is not to solve this conundrum, nor to try to convince you how you should consider numbers.
+However, in IPLD, we usually do consider integers to be [infinite scalars](#infinite-scalars),
+with the caveat that libraries processing data which exceeds the boundaries they can process should error and halt explicitly.
+(In IPLD, we have the good fortune that this definition is unusually easy to stick to:
+since we only carry data around, and don't provide mathematical operations on it,
+it's easy to expect that data which was processable on the way in will remain processable throughout the time we handle it.)
+
+
+Types in IPLD Schemas
+---------------------
+
+It probably won't surprise to you hear that IPLD Schema's core features
+map very closely to the categories of types we've discussed:
+
+- [enums](#enums) are `enum`.
+- [sum types](#sum-types) are `union`.
+- [product types](#product-types) are `struct`.
+- [infinite recursives](#infinite-recursives) appear as `list` and `map`.
+- [infinite scalars](#infinite-scalars) appear as `string`, `bytes`, and `int`.
+- [sum types](#sum-types) specifically for [maybe](#maybe-types) show up as `optional` and `nullable`.
+	- (there are two of them because each creates a sum with a different [unit type](#unit-types)!)
+
+IPLD Schemas don't have a concept of [interfaces](#open-vs-closed-membership).
+At least, *not exactly* -- not *within* the schema.
+Instead, the *whole Schema* is treated as an interface:
+since you can take some data and take a schema, and attempt to unify them to get typed data if the schema "matches"...
+you can use a series of Schemas in the same way as you'd use a series of interfaces.
+(IPLD Schemas are effectively [*structural*](explicit-versus-structural-interfaces) interfaces.)
+
+
+Different sizes of Infinity
+---------------------------
+
+// todo: not quite sure what to say about this.  the answer isn't "give up", that's for sure.
```

Comment:
Is there much that you want to say here beyond what you've already said above, e.g. with "maybe map"?